golemfactory/yagna
# zksync driver deposit (`enter` CLI command)

Fixes [PAY-64](https://golemproject.atlassian.net/browse/PAY-64)

Steps to test the command:
- prepare `.env` file
- provide valid eth node address for `ZKSYNC_RINKEBY_GETH_ADDR` key (e.g. infura) in the `.env` 
- run the service `cargo run service run`
- open new terminal window and there run the following commands
- `cargo run payment fund --driver erc20`
- `cargo run payment init --sender --driver zksync` (will fail but unlocks the wallet of the zksync account)
- `cargo run payment enter --amount 5 --driver=zksync` starts a deposit transaction. When successful will provide a link to track the transaction on the block explorer, e.g. [example1](https://rinkeby.zkscan.io/explorer/transactions/0xf560fbd6c2201e242c4353ab9cbdd96bb4270a0fb41a6842af8571b165487ea5) [example2](https://rinkeby.zkscan.io/explorer/transactions/3fff40f2f3fb55b2159ebc1f212fdd37cf1e45c6b58503009e484d51a119e133)
- when the transaction completes, funds can be seen with `cargo run payment status --driver zksync`

On the block explorer for Ethereum testnet you should see also 2 new transactions:
- allowing ERC20 token deposit on zksync contract [such this one](https://rinkeby.etherscan.io/tx/0xa94576c5d4ca4f4fcd5cb95f622b2e89305acf4aa8ecd4e1317f100134fb83a7)
- deposit transaction from L1 to ZkSync contract [example](https://rinkeby.etherscan.io/tx/0x3fff40f2f3fb55b2159ebc1f212fdd37cf1e45c6b58503009e484d51a119e133)
